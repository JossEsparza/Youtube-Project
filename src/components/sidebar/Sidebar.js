import { useContext } from 'react';
import { NavLink } from 'react-router-dom';
import { ToggleContext } from '../../state/ToggleContext';
import './Sidebar.scss';

export const Sidebar = () => {
  const { dispatch } = useContext(ToggleContext);
  const handleChange = () => {
    dispatch({
      type: 'set-input',
      payload: 'changuitos',
    });
  };
  return (
    <nav className="sidebar_container">
      <NavLink
        to="/"
        className={(isActive) => (isActive ? 'nav-active' : '')}
        onClick={handleChange}
        exact
      >
        Home
      </NavLink>
      <NavLink
        to="/favorites"
        className={(isActive) => (isActive ? 'nav-active' : '')}
        exact
      >
        Favorites
      </NavLink>
    </nav>
  );
};
