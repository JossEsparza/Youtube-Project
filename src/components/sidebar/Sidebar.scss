.dark {
  .sidebar_container {
    background-color: #19171c;
    border-right: 1.5px solid #3c3a3e;
  }
  a {
    color: #a3a3a6;
    &:hover {
      font-weight: bold;
      color: #cbcccd;
    }
  }
  a.nav-active {
    color: #cbcccd;
    font-weight: bold;
  }
}

.light {
  .sidebar_container {
    background-color: #f8f8fa;
  }
  a {
    color: #a3a3a6;
    &:hover {
      font-weight: bold;
      color: #383f44;
    }
  }
  a.nav-active {
    color: #383f44;
    font-weight: bold;
  }
}

.sidebar_container {
  display: flex;
  flex-direction: column;
  width: 167px;
  height: 100vh;
  min-width: 167px;
}

a {
  margin-bottom: 10px;
  margin-top: 10px;
  text-align: center;
  text-decoration: none;
}
