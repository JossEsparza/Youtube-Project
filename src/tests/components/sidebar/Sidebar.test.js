import { mount } from 'enzyme';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { ToggleContext } from '../../../state/ToggleContext';
import { Sidebar } from '../../../components/sidebar/Sidebar.js';

describe('Tests the Sidebar', () => {
  test('should be displayed correctly', () => {
    const globalReducer = {
      inputValue: 'changuitos',
      currentToggle: false,
      isEnabled: false,
    };
    const wrapper = mount(
      <ToggleContext.Provider
        value={{
          globalReducer,
        }}
      >
        <Router history={createMemoryHistory()}>
          <Sidebar />
        </Router>
      </ToggleContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
