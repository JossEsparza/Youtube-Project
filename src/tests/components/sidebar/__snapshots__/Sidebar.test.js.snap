// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Sidebar should be displayed correctly 1`] = `ReactWrapper {}`;
